src/test/java/com/urswolfer/gerrit/client/rest/http/projects/ProjectsParserTest.java

```diff
@@ -78,4 +84,17 @@ public void testParseSingleProjectInfo() throws Exception {
             .withState(ProjectState.ACTIVE)
             .get());
     }
+
+    @Test
+    public void testGenerateProjectInput() throws Exception {
+        ProjectInput projectInput = new ProjectInput();
+        projectInput.name = "MyProject";
+        projectInput.description = "This is a demo project.";
+        projectInput.submitType = SubmitType.CHERRY_PICK;
+        projectInput.owners = Collections.singletonList("MyProject-Owners");
+
+        String actualProjectInput = Files.readFile(getFile("project_input.json")).replaceAll("\n","");
+        Truth.assertThat(projectsParser.generateProjectInput(projectInput))
```

Comment:
Yes, that was the main reason for adding JSON conversion to the parser.

Yes, comparing it to a plain string is pretty unsafe; I had to mangle the json around to get this test to work. That said, the test itself is just ensuring gson works. I'll change this test to deserialize the string output into a separate ProjectInput object, check that the fields are alright.

I don't understand what you mean here; I mocked out the parser in the client test.